Library_Engine/Query/ValidatePath.cs

```diff
@@ -0,0 +1,82 @@
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using BH.oM.Base;
+using BH.oM.Base.Attributes;
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+using BH.oM.Versioning;
+
+namespace BH.Engine.Library
+{
+    public static partial class Query
+    {
+        /***************************************************/
+        /**** Public Methods                            ****/
+        /***************************************************/
+
+        [Description("Validates that the provided string is a valid full library path, and atempts to upgrade the path if it is not. Returns the input path if valid, or failing to upgrade. Returns the upgraded path if an upgrade is possible.")]
+        [Input("fullLibraryName", "The full library path to the particular Library to validate. Only full paths supported, not super paths or partial paths to libraries.")]
+        [Output("path", "Returns the input path if valid or failing to upgrade. Returns the upgraded path if able to upgrade.")]
+        public static string ValidatePath(string fullLibraryName)
+        {
+            List<string> fullLibraryNames = LibraryNames();
+            if (fullLibraryNames.Contains(fullLibraryName))
+                return fullLibraryName;
+
+            string newName = fullLibraryName;
+            string upgradedName;
+            Dictionary<string, string> datasetUpgrades = Engine.Versioning.Query.DatasetToNewPaths();
+            while (datasetUpgrades.TryGetValue(newName, out upgradedName))
+            {
+                newName = upgradedName;
+            }
+
+            if (newName == null || !fullLibraryNames.Contains(newName))
+            {
+                Engine.Base.Compute.RecordError($"The dataset {fullLibraryName} is not a valid full path library and no valid upgrade could be found.");
+                return fullLibraryName;
+            }
+            else
+            {
+                string newVersion = Engine.Base.Query.BHoMVersion();
+                string oldVersion = "?.?";
```

Comment:
I take it we don't have a way to work this out at all? Or not currently at least?

Reply:
Discussed on call, think it is possible to fix this with slight tweak of how the code runs.